test/xbgpu/test_bsend.py

```diff
@@ -0,0 +1,228 @@
+
+        Parameters
+        ----------
+        outputs
+            Fixture
+        queues
+            List of :class:`spead2.InprocQueue` used to transmit heaps
+            in :meth:`_send_data`.
+        channel_offset, n_channels_per_substream, n_spectra_per_heap, heap_timestamp_step
+            Variables declared by the calling unit test to verify
+            transmitted data.
+        """
+        out_config = spead2.recv.StreamConfig(max_heaps=100)
+        out_tp = spead2.ThreadPool()
+        for queue in queues:
+            stream = spead2.recv.asyncio.Stream(out_tp, out_config)
+            stream.add_inproc_reader(queue)
+
+            # Wait for the first packet to arrive - it is expected to be the
+            # SPEAD descriptor.
+            ig = spead2.ItemGroup()
+            heap = await stream.get()
+            # TODO: No checks on the count sequence yet
+            items = ig.update(heap)
+            assert items == {}, "This heap contains item values not just the expected descriptors."
+
+            # Check the data heaps
+            zero_data = np.zeros(shape=(n_channels_per_substream, n_spectra_per_heap, COMPLEX), dtype=np.int8)
+            for i in range(TOTAL_DATA_HEAPS_PER_SUBSTREAM):
+                heap = await stream.get()
+                items = ig.update(heap)
+                assert set(items.keys()) == {"timestamp", "frequency", "bf_raw"}
+                assert items["timestamp"].id == TIMESTAMP_ID
+                assert items["timestamp"].value == i * heap_timestamp_step
+                assert items["frequency"].id == FREQUENCY_ID
+                assert items["frequency"].value == channel_offset
+                assert items["bf_raw"].id == BF_RAW_ID
+                assert items["bf_raw"].value.shape == (n_channels_per_substream, n_spectra_per_heap, COMPLEX)
+                assert items["bf_raw"].value.dtype == np.int8
+                np.testing.assert_equal(items["bf_raw"].value, zero_data)
+
+    @pytest.mark.combinations(
+        "num_channels, num_spectra_per_heap",
+        test_parameters.num_channels,
+        test_parameters.num_spectra_per_heap,
+    )
+    async def test_send_simple(
+        self,
+        context: AbstractContext,
+        num_channels: int,
+        num_spectra_per_heap: int,
+        outputs: Sequence[BOutput],
+        time_converter: TimeConverter,
+        sensors: SensorSet,
+    ) -> None:
+        """
+        Test :class:`katgpucbf.xbgpu.bsend.BSend`.
+
+        This test transmits a number of heaps from a BSend object over a spead2
+        in-process transport. The received heaps are then checked.
+
+        This test does not generate random data as it will take much more compute
+        to check that the random data is received correctly.
+
+        .. todo::
+
+            Update this test to make use of mock_{send, recv}_stream fixtures.
+            Perhaps just mock_send_stream, as mock_recv_stream might require
+            more refactoring.
```

Comment:
Why? At this level, a custom stream_factory is the appropriate way to make use of the inproc transport.

Reply:
Too right, I think I was too focused on the whole 'use the fixtures we have'. Also somehow concluded test_xsend.py was out of date since it hasn't been touched for a while. 

Will remove this `todo`.